File: Challenges/Kidus/js/RemoveDuplicatesfromanArray.js
function removeDuplicates(arr) {
    let uniqueArr = []; // Create an empty array to store unique values
    for (let i = 0; i < arr.length; i++) {
        if (!uniqueArr.includes(arr[i])) { // If the value is not already in uniqueArr
            uniqueArr.push(arr[i]); // Add it to uniqueArr
        }
    }
    return uniqueArr;
}

console.log(removeDuplicates([1, 2, 2, 3, 4, 4, 5])); 
// Output: [1, 2, 3, 4, 5]

console.log(removeDuplicates(["apple", "banana", "apple", "orange", "banana"])); 
// Output: ["apple", "banana", "orange"]

// Using a set

function removeDuplicatesUsingSet(arr) {
    let uniqueSet = new Set(arr); // Create a Set from the array
    return [...uniqueSet]; // Convert the Set back to an array
}

File: Challenges/Kidus/python/RemoveDuplicatesfromanArray.py
def removeDuplicates(arry):
    """
    Remove duplicates from a given array using a dictionary.
    """
    # Create an empty dictionary to store unique elements
    unique_dict = {}
    
    # Iterate through the array
    for element in arry:
        # Check if the element is already in the dictionary
        if element not in unique_dict:
            # Add the element to the dictionary
            unique_dict[element] = True
    
    # Convert the dictionary back to a list
    unique_list = list(unique_dict.keys())
    
    return unique_list

# Test the function

print(removeDuplicates([1, 2, 3, 2, 1, 4, 5]))  # Output: [1, 2, 3, 4, 5]
print(removeDuplicates(["apple", "banana", "apple", "cherry", "banana"]))  # Output: ["apple", "banana", "cherry"]

print(removeDuplicates([1, 2, 3, 4, 5, 5, 5, 5, 5]))  # Output: [1, 2, 3, 4, 5]